lib/auth/webauthnwin/webauthn_windows.go

```diff
@@ -0,0 +1,633 @@
+		return nil, trace.Wrap(err)
+	}
+	if ret != 0 {
+		return nil, trace.Wrap(getErrorName(ret))
+	}
+	if out == nil {
+		return nil, errors.New("unexpected nil response from MakeCredential")
+	}
+
+	// Note that we need to copy bytes out of `out` if we want to free object.
+	// That's why bytesFromCBytes is used.
+	// We don't care about free error so ignore it explicitly.
+	defer func() { _ = freeCredentialAttestation(out) }()
+
+	credential := bytesFromCBytes(out.CbCredentialId, out.PbCredentialId)
+
+	return &wanlib.CredentialCreationResponse{
+		PublicKeyCredential: wanlib.PublicKeyCredential{
+			Credential: wanlib.Credential{
+				ID:   base64.RawURLEncoding.EncodeToString(credential),
+				Type: string(protocol.PublicKeyCredentialType),
+			},
+			RawID: credential,
+		},
+		AttestationResponse: wanlib.AuthenticatorAttestationResponse{
+			AuthenticatorResponse: wanlib.AuthenticatorResponse{
+				ClientDataJSON: jsonEncodedCD,
+			},
+			AttestationObject: bytesFromCBytes(out.CbAttestationObject, out.PbAttestationObject),
+		},
+	}, nil
+}
+
+func freeCredentialAttestation(in *webauthnCredentialAttestation) error {
+	_, _, err := procWebAuthNFreeCredentialAttestation.Call(
+		uintptr(unsafe.Pointer(in)),
+	)
+	if err != syscall.Errno(0) {
+		return err
+	}
+	return nil
+}
+
+func freeAssertion(in *webauthnAssertion) error {
+	_, _, err := procWebAuthNFreeAssertion.Call(
+		uintptr(unsafe.Pointer(in)),
+	)
+	if err != syscall.Errno(0) {
+		return err
+	}
+	return nil
+}
+
+// checkIfDLLExistsAndGetAPIVersionNumber checks if dll exists and tries to load
+// it's version via API call. This function makes sure to not panic if dll is
+// missing.
+func checkIfDLLExistsAndGetAPIVersionNumber() (int, error) {
+	if err := modWebAuthn.Load(); err != nil {
+		return 0, err
+	}
+	if err := procWebAuthNGetApiVersionNumber.Find(); err != nil {
+		return 0, err
+	}
+	ret, _, err := procWebAuthNGetApiVersionNumber.Call()
+	if err != syscall.Errno(0) {
+		return 0, err
+	}
+	return int(ret), nil
+}
+
+func getErrorName(in uintptr) error {
+	ret, _, err := procWebAuthNGetErrorName.Call(
+		uintptr(int32(in)),
+	)
+	if err != syscall.Errno(0) {
+		return fmt.Errorf("webauthn error code %v", in)
+	}
+	errString := windows.UTF16PtrToString((*uint16)(unsafe.Pointer(ret)))
+	return fmt.Errorf("webauthn error code %v: %v", in, errString)
+}
+
+func isUVPlatformAuthenticatorAvailable() (bool, error) {
+	var out uint32
+	ret, _, err := procWebAuthNIsUserVerifyingPlatformAuthenticatorAvailable.Call(
+		uintptr(unsafe.Pointer(&out)),
+	)
+	if err != syscall.Errno(0) {
+		return false, err
+	}
+	if ret != 0 {
+		return false, getErrorName(ret)
+	}
+	return out == 1, nil
+}
+
+func (c client) getAssertionOptionsVersion() uint32 {
+	// Mapped based on:
+	// https://github.com/microsoft/webauthn/blob/7ab979cc833bfab9a682ed51761309db57f56c8c/webauthn.h#L36-L96
+	switch c.webauthnAPIVersion {
+	case 1, 2:
+		return 4
+	case 3:
+		return 5
+	default: // >= 4
+		return 6
```

Comment:
We have other threads about this, but just so it doesn't lost: let's follow up on whether we need to adjust versions or if we can just always set `6`.

Same for similar logic in other places.